### DISCUSSION TITLE
Strict type checking in query parameters causes validation errors for integer inputs

### DISCUSSION THREAD

**Discussion by Phinnik** [fastapi/fastapi#11631] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
import fastapi


app = fastapi.FastAPI()


@app.get("/example/{user_id}")
def example_get(
    user_id: Annotated[int, fastapi.Path(gt=0)],
    count: Annotated[int, fastapi.Query(gt=0, strict=True)] = 10,
):
    return f"{user_id=}, {count=}"
```

### Description

When using the strict=True parameter in FastAPI's Query for an integer type, the endpoint rejects valid integer inputs with a validation error, stating that the input should be a valid integer.
query: `http://0.0.0.0:8080/example/1?count=10`
response:
```json
{
    "detail": [
        {
            "type": "int_type",
            "loc": [
                "query",
                "count"
            ],
            "msg": "Input should be a valid integer",
            "input": "10"
        }
    ]
}
```

### Operating System

Linux

### FastAPI Version

0.111.0

### Pydantic Version

2.7.1

### Python Version

Python 3.11.9

**Comment by YuriiMotov** [comment, marked as answer]
All the non-Body parameters FastAPI receives are initially strings. And FastAPI passes these strings as a parameters to Pydantic validators.
If you specify `strict=True` Pydantic will not accept `str` instead of `int`.

Why do you need this 'strictness'?
In your example the worse thing that can happen without `strict=True` is that it will accept `10.0` as valid integer parameter. But it will still not let you pass `10.1`
Is it a problem? Or, am I missing something?
